exaudfclient/base/exaudflib/impl/check.cc

```diff
@@ -0,0 +1,77 @@
+#include "exaudflib/impl/check.h"
+#include "exaudflib/impl/socket_info.h"
+#include <pthread.h>
+#include <unistd.h>
+#include <iostream>
+
+#include "debug_message.h"
+
+namespace exaudflib {
+    namespace check {
+        static bool keep_checking = true;
```

Comment:
ok, I see this is actually quite sketchy, in really this should be an std::atomic

Reply:
Indeed. Interesting that this never caused an issue, it isn't even declared as volatile.